### PR TITLE
group creation refurb

### PR DESCRIPTION
**Pull request by patosullivan**
Fixes TLON-2714
Fixes TLON-2312
Fixes TLON-2733

Implements the new group/DM creation/join flow, which all starts with a single sheet now. I made sure this all works on web now, as well (by switching to using basic react state for view switching in AddGroupSheet).

Removed the FloatingAddButton since it's no longer necessary (it was only used in ChatList for triggering either the new dm or new group sheets).

Updated FloatingActionButton to match the new rounded style.

Broke `performGroupActions` out of useChannelNavigation, it now lives in useGroupActions.

Fixed an issue where the Invite Users sheet wasn't accessible from the ChatListOptions within the GroupChannels view.

Also rewrote InviteUsersSheet so that it uses the Modal component. Before this rewrite, the app would crash with a complaint about RCTView not appearing as a descendant of RCTShadowView when rendered in the EmptyChannelComponent.

### CODE REVIEW

**Comment by jamesacklin** [comment]
Thank you! Only visual nits.

1) Let's use medium-weight (500?) for the "New Message" header and track the spacing in a bit. Also if there's a way to get all the left edges to line up, great.

Screenshot 2024-09-17 at 1 51 46 PM

2) "Select Members" here is also too bold. 500 should do it.

Simulator Screenshot - iPhone 15 Pro - 2024-09-17 at 13 52 36


3) Investigate using the Input components from the UI library. The border here will end up being much lighter and the label will be better aligned. I'm pretty sure that component will also take validator rules in case we want to simplify anything. 

The disabled button on this screen is also much too dark. We should crib from somewhere else in the app for a disabled state.

Screenshot 2024-09-17 at 1 52 50 PM

The transitions between states in the sheet look OK, I think. I'm not sure we have sheet content state transitions elsewhere (group actions/admin, etc).

**Review by dnbrwstr — CHANGES_REQUESTED** [review]
Overall looking good; to build on James' comment above: all of this (except the actual contact list) should be doable w/ standard action sheet components/ follow the same basic structure of other action sheets. If that's not doable, I'm curious about where.

Edit: would recommend looking at `CreateChannelSheet` for reference.

**Comment by jamesacklin** [comment]
I see this error when creating a group:

image

Otherwise looks good visually.

**Comment by patosullivan** [comment]
> I see this error when creating a group:
> 
> image
> 
> Otherwise looks good visually.

this is fixed in the latest commit

**Review by dnbrwstr — COMMENTED** [review]
Nice, a couple nits here, and also:
- Horizontal margins are thicker on this sheet vs other sheets.
- Additional items should scroll under search input, not just the contact names
- iOS "new group" transition is a little wonky -- incoming views should slide over old views; right now it's doing a kind of crisscross thing (or, really, we should just open in a new screen like the other interactions do
- I'm seeing a lot of "invalid snap points" warnings

**Comment by patosullivan** [comment]
> Nice, a couple nits here, and also:
> 
> * Horizontal margins are thicker on this sheet vs other sheets.
> * Additional items should scroll under search input, not just the contact names
> * iOS "new group" transition is a little wonky -- incoming views should slide over old views; right now it's doing a kind of crisscross thing (or, really, we should just open in a new screen like the other interactions do
> * I'm seeing a lot of "invalid snap points" warnings

The snap points warning is happening because I've set `snapPoints` to ['90%'] and I haven't set `snapPointsMode` to "percent". If I don't set `snapPoints` then the sheet doesn't pop up to an appropriate height:
Simulator Screenshot - iPhone 15 - 2024-09-18 at 13 38 35

If I do set `snapPoints` to 90% with `snapPointsMode` set to "percent", it instead takes up the whole screen (and weirdly cuts off the bottom content of the sheet):
Simulator Screenshot - iPhone 15 - 2024-09-18 at 13 39 45

So I ended up doing what we're doing in ProfileSheet, setting snapPoints without changing the mode, which does seem to actually respect the snapPoints setting.

Which additional items are we expecting under the list of contacts?

File: packages/app/hooks/useGroupActions.native.tsx
[the remark this answers is not in the document]
Comment (patosullivan) [reply packages/app/hooks/useGroupActions.native.tsx:1]: Not sure I understand, unless you mean we get rid of this hook and just define `performGroupActions` in the relevant controllers and pass it as a prop instead?

[the remark this answers is not in the document]
Comment (dnbrwstr) [reply packages/app/hooks/useGroupActions.native.tsx:1]: Just that the only difference between the web and native versions are the actual `navigate` calls -- could we instead move the navigation calls to something like a useGroupNavigation hook and have a single useGroupActions hook so as not to duplicate the action parsing + data access logic?

**Comment by dnbrwstr** [comment]
> Which additional items are we expecting under the list of contacts?

The idea is that the "New group" and "Join group" controls are part of the list content, so they should scroll along with the content below.

**Review by dnbrwstr — COMMENTED** [review]
Great, this is looking good. Two last bits:
- I'm still seeing the snap point warning? Could that be something on my end?
- Can "new group" open on a new screen? Or at minimum can we fix the transition so that new view slides on top of old view rather than old view sliding out to right as new view slides in from right?

**Comment by patosullivan** [comment]
> Great, this is looking good. Two last bits:
> 
> * I'm still seeing the snap point warning? Could that be something on my end?
> * Can "new group" open on a new screen? Or at minimum can we fix the transition so that new view slides on top of old view rather than old view sliding out to right as new view slides in from right?

See my comments above about the snap points issue. Yeah, I can make a screen for new group.

File: packages/ui/src/components/AddGroupSheet.tsx [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+    [goToScreen, onGoToDm, groupInvitees, currentScreen, navigateToFindGroups]
+  );
+
+  return (
+    <ActionSheet
+      disableDrag={screenScrolling}
+      moveOnKeyboardChange
+      open={open}
+      onOpenChange={dismiss}
+      snapPoints={['90%']}
```
Comment (dnbrwstr) [inline packages/ui/src/components/AddGroupSheet.tsx:184]: This fixes the snapPoints warning for me

```suggestion
      snapPoints={[90]}
      snapPointsMode="percent"
```

**Review by dnbrwstr — APPROVED** [review]
Lgtm! Appreciate the edits.

**Pull request merged**